Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -6,33 +6,60 @@
 
 import Foundation
 
+/// Feature-agnostic SDK configuration.
+internal typealias CoreConfiguration = FeaturesConfiguration.Common
+
+/// Feature-agnostic set of dependencies powering Features storage, upload and event recording.
+internal typealias CoreDependencies = FeaturesCommonDependencies
+
+/// A shim interface for uniforming V1 Features and letting their generic initialization in `DatadogCore`.
+internal protocol V1Feature {
+    /// The configuration specific to this Feature.
+    /// In V2 this will likely become a part of the public interface for the Feature module.
+    associatedtype Configuration
+
+    init(
+        storage: FeatureStorage,
+        upload: FeatureUpload,
+        configuration: Configuration,
+        commonDependencies: FeaturesCommonDependencies
+    )
+}
+
 /// Core implementation of Datadog SDK.
 ///
-/// The core provides a storage and upload mechanism for each registered
-/// feature based on their respective configuration.
+/// The core provides a storage and upload mechanism for each registered Feature
+/// based on their respective configuration.
```

Comment:
Here, I started using "**F**eature" (Logging / RUM / Tracing) instead of "**f**eature" (e.g. sampling or mapping) to distinguish it in doc comments. Does it make sense @maxep?